qa/rolling-upgrade/src/test/java/org/opensearch/knn/bwc/IndexingIT.java

```diff
@@ -5,39 +5,53 @@
 
 package org.opensearch.knn.bwc;
 
-import static org.opensearch.knn.TestUtils.*;
+import java.io.IOException;
+
+import static org.opensearch.knn.TestUtils.NODES_BWC_CLUSTER;
 
 public class IndexingIT extends AbstractRollingUpgradeTestCase {
     private static final String TEST_FIELD = "test-field";
     private static final int DIMENSIONS = 5;
+    private int docId = 0;
     private static final int K = 5;
-    private static final int ADD_DOCS_CNT = 10;
+    private static final int NUM_DOCS = 10;
+    private int totalDocsCount = 0;
 
-    public void testKnnDefaultIndexSettings() throws Exception {
+    public void testKNNDefaultIndexSettings() throws Exception {
         waitForClusterHealthGreen(NODES_BWC_CLUSTER);
         switch (getClusterType()) {
             case OLD:
                 createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
-                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_CNT);
+                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
                 break;
             case MIXED:
                 if (isFirstMixedRound()) {
-                    validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 10, K);
-                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 10, ADD_DOCS_CNT);
-                    validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 20, K);
+                    totalDocsCount = NUM_DOCS;
+                    docId = NUM_DOCS;
                 } else {
-                    validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 20, K);
-                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 20, ADD_DOCS_CNT);
-                    validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 30, K);
+                    totalDocsCount = 2 * NUM_DOCS;
+                    docId = 2 * NUM_DOCS;
                 }
+                validateKNNIndexingOnUpgrade();
                 break;
             case UPGRADED:
-                validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 30, K);
-                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 30, ADD_DOCS_CNT);
-                validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 40, K);
+                totalDocsCount = 3 * NUM_DOCS;
+                docId = 3 * NUM_DOCS;
+                validateKNNIndexingOnUpgrade();
+
                 forceMergeKnnIndex(testIndex);
-                validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 40, K);
+                validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, totalDocsCount, K);
+
                 deleteKNNIndex(testIndex);
         }
     }
+
+    // validation steps for indexing after upgrading each node from old version to new version
+    public void validateKNNIndexingOnUpgrade() throws IOException {
```

Comment:
Hmmm I think this should be configurable to take arguments instead of relying on member variables to prevent any shared state between tests. @martin-gaievski what are your thoughts?

Reply:
agree, it makes more sense to inject params